tests/token-guard.ts

```diff
@@ -411,123 +320,320 @@ describe("token-guard", () => {
+        let metadata: web3.PublicKey;
 
-      const instructions = await exchange(
-        provider.connection,
-        program,
-        tokenGuardState.id,
-        sender.publicKey,
-        sender.publicKey,
-        gatekeeperNetwork.publicKey,
-        exchangeAmount,
-        {
-          tokenAccount: senderMembershipTokenATA,
-        }
-      );
+        const nftMinter = web3.Keypair.generate();
+        const nftMinterProvider = new Provider(
+          provider.connection,
+          new anchor.Wallet(nftMinter),
+          {}
+        );
 
-      await sendTransactionFromSender(instructions);
+        before("Mint an NFT", async () => {
+          await fund(nftMinter.publicKey);
+          sandbox.stub(axios, "get").resolves({ data: metadataJson });
+          // the minter has to be in the creators array
+          metadataJson.properties.creators[0].address =
+            nftMinter.publicKey.toBase58();
+
+          const metadataUri = "somewhere";
+          const response = await actions.mintNFT({
+            connection: nftMinterProvider.connection,
+            wallet: nftMinterProvider.wallet,
+            uri: metadataUri,
+            maxSupply: 1,
+          });
+
+          mint = response.mint;
+          metadata = response.metadata;
+        });
+
+        it("should initialize a tokenGuard that requires presentation of an NFT", async () => {
+          tokenGuardState = await initialize(
+            program,
+            provider,
+            gatekeeperNetwork.publicKey,
+            recipient.publicKey,
+            undefined,
+            undefined,
+            undefined,
+            {
+              key: nftMinter.publicKey,
+              strategy: "NFT-UA",
+            }
+          );
+        });
+
+        it("should not let someone without the membership token exchange", async () => {
+          const shouldFail = exchange(
+            provider.connection,
+            program,
+            tokenGuardState.id,
+            sender.publicKey,
+            sender.publicKey,
+            gatekeeperNetwork.publicKey,
+            exchangeAmount
+          );
+
+          return expect(shouldFail).to.be.rejectedWith(
+            /Membership token account not found/
+          );
+        });
+
+        it("should let someone with the token exchange", async () => {
+          const nft = new Token(
+            provider.connection,
+            mint,
+            TOKEN_PROGRAM_ID,
+            recipient
+          );
+          const minterATA = await Token.getAssociatedTokenAddress(
+            ASSOCIATED_TOKEN_PROGRAM_ID,
+            TOKEN_PROGRAM_ID,
+            mint,
+            nftMinter.publicKey
+          );
+
+          const senderMembershipTokenATA = await nft.createAccount(
+            sender.publicKey
+          );
+          await nft.transfer(
+            minterATA,
+            senderMembershipTokenATA,
+            nftMinter,
+            [],
+            1
+          );
```

Comment:
I _think_ you should be able to mint directly to the `senderMembershipTokenATA` and avoid the need to create the `minterATA` and this transfer altogether. I could be wrong, though.

Reply:
That would be great. I didn't see any way through the @metaplex/js

```
const response = await actions.mintNFT({
            connection: nftMinterProvider.connection,
            wallet: nftMinterProvider.wallet,
            uri: metadataUri,
            maxSupply: 1,
          });
```